lua/lspconfig/util.lua

```diff
@@ -272,6 +272,7 @@ function M.server_per_root_dir_manager(make_config)
           { { uri = vim.uri_from_fname(root_dir), name = root_dir } },
           { {} }
         )
+        vim.lsp.buf_notify(0, 'workspace/didChangeWorkspaceFolders', params)
```

Comment:
`client.notify('workspace/didChangeWorkspaceFolders', params)` and 
        `vim.lsp.buf_notify(0, 'workspace/didChangeWorkspaceFolders', params)` are both not working in gopls

Reply:
in air project open another go project still get an error right?

Reply:
yes

Reply:
hmm. there already have `return client.id`

Reply:
this return is in
function  ` function manager.add` 

`return client.id` is in outer function

Reply:
I think you need this https://github.com/golang/tools/blob/master/gopls/doc/workspace.md

Reply:
looks like recently go most suggest use the go work and in my test it works fine .

Reply:
thank you, so it means I need to put the project into the same root path and create a go.work for it.
but sadly my company's go version is 1.17 which is not supported go.work, it still has issues for me without go.work.
in the previous version which is for each workspace to start a new LSP server works well, but in the master branch the behavior is broken.

Reply:
> in the previous version which is for each workspace to start a new LSP server works well, but in the master branch the behavior is broken.

This is a mistake. When you open a file in a project it spawns a new process. These items should be parsed by a language server instance.

a useful ref  https://github.com/golang/vscode-go/issues/2112 . there has a setting maybe you can use in gopls